java/src/org/openqa/selenium/grid/node/config/SessionCapabilitiesMutator.java

```diff
@@ -56,33 +57,137 @@ public Capabilities apply(Capabilities capabilities) {
     }
 
     String browserName = capabilities.getBrowserName().toLowerCase();
-    if (!BROWSER_OPTIONS.containsKey(browserName)) {
-      return capabilities;
-    }
-
     String options = BROWSER_OPTIONS.get(browserName);
-    if (!slotStereotype.asMap().containsKey(options)) {
-      return capabilities;
+    if (slotStereotype.asMap().containsKey(options) && capabilities.asMap().containsKey(options)) {
+
+      @SuppressWarnings("unchecked")
+      Map<String, Object>
+        stereotypeOptions =
+        (Map<String, Object>) slotStereotype.asMap().get(options);
+
+      @SuppressWarnings("unchecked")
+      Map<String, Object> capsOptions = (Map<String, Object>) capabilities.asMap().get(options);
+
+      // Merge top level capabilities, excluding browser specific options.
+      // This will overwrite the browser options too, but it does not matter since we tackle it separately just after this.
+      Map<String, Object> toReturn = new HashMap<>(slotStereotype.merge(capabilities).asMap());
+
+      // Merge browser specific stereotype and capabilities options
+      switch (browserName) {
+        case "chrome":
+        case "microsoftedge":
```

Comment:
In some places in the Java code I see we consider `MicrosoftEdge` and `msedge`, maybe consider it here as well?